### PR TITLE
Practice problems

### PR DESCRIPTION
**Pull request by ceciliachan1979**

### CODE REVIEW

File: Misc/Practice/Practice.tex [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+\subsubsection*{Solution}
+This is a rather brute force solution, I guess better solution exists.
+
+\begin{align*}
+  p(x) + 1 &| (x - 1)^3                                                                           \\
+  p(x) + 1 &= (x - 1)^3(ax^2 + bx + c)                                                            \\
+           &= (x^3 - 3x^2 + 3x - 1)(ax^2 + bx + c)                                                \\
+           &= ax^5 + (-3a + b)x^4 + (3a - 3b + c)x^3 + (-a + 3b - 3c)x^2 + (-bx + 3c)x - c        \\
+      p(x) &= ax^5 + (-3a + b)x^4 + (3a - 3b + c)x^3 + (-a + 3b - 3c)x^2 + (-bx + 3c)x + (-c - 1)
```
Comment (AoiYamada) [inline Misc/Practice/Practice.tex:25]:
```suggestion
           &= ax^5 + (-3a + b)x^4 + (3a - 3b + c)x^3 + (-a + 3b - 3c)x^2 + (-b + 3c)x - c        \\
      p(x) &= ax^5 + (-3a + b)x^4 + (3a - 3b + c)x^3 + (-a + 3b - 3c)x^2 + (-b + 3c)x + (-c - 1)
```
typo?

File: Misc/Practice/Practice.tex [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+\end{align*}
+
+\begin{align*}
+  p(x) - 1 &| (x + 1)^3                                                                       \\
+  p(x) - 1 &= (x + 1)^3(dx^2 + ex + f)                                                        \\
+           &= (x^3 + 3x^2 + 3x + 1)(dx^2 + ex + f)                                            \\
+           &= dx^5 + (3d + e)x^4 + (3d + 3e + f)x^3 + (d + 3e + 3f)x^2 + (ex + 3f)x + f       \\
+      p(x) &= dx^5 + (3d + e)x^4 + (3d + 3e + f)x^3 + (d + 3e + 3f)x^2 + (ex + 3f)x + (f + 1) \\
```
Comment (AoiYamada) [inline Misc/Practice/Practice.tex:33]:
```suggestion
           &= dx^5 + (3d + e)x^4 + (3d + 3e + f)x^3 + (d + 3e + 3f)x^2 + (e + 3f)x + f       \\
      p(x) &= dx^5 + (3d + e)x^4 + (3d + 3e + f)x^3 + (d + 3e + 3f)x^2 + (e + 3f)x + (f + 1) \\
```

**Review by Andy-Berkshire — CHANGES_REQUESTED** [review]
Line 61, replace 5x^3/8 with 10x^3/8

File: Misc/Practice/Practice.tex [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+And therefore we determine $ C = \frac{1}{24} $ by simply making sure when $ x = 4 $ the output is 16. 
+
+The observation is that the coefficients seems to be $ \frac{1}{k!} $, it would be great if that's true because then we know how to construct these polynomial for arbitrary $ k $.
+
+To show that, we will prove by induction that $ p_k(k+1) = 2^{k + 1} - 1 $, it is obviously true for $ k = 0 $ as $ p_0(1) = 1 = 2^{1} - 1 $.
+
+Suppose we know $ p_k(k+1) = 2^{k + 1} - 1 $, then
+
+\begin{align*}
+  p_{k+1}(k+1) &= p_k(x + 1) + Cx(x-1)\cdots(x - k) \\
```
Comment (Andy-Berkshire) [inline Misc/Practice/Practice.tex:98]: 2^{k+1} = p_{k+1}(k+1) &= \left. p_k(x) + Cx(x-1)\cdots(x - k) \right|_{x=k+1}

File: Misc/Practice/Practice.tex [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+               &= 2^{k + 1} - 1 + C (k + 1)!        \\
+               &= 2^{k + 1} &\text{As required}
```
Comment (Andy-Berkshire) [inline Misc/Practice/Practice.tex:100]: delete this line

File: Misc/Practice/Practice.tex [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  c &= \frac{-8}{8} \\
+  d &= \frac{-3}{8} \\
+  e &= \frac{ 9}{8} \\
+  f &= \frac{-8}{8}
+\end{align*}
+
+Substitute and expand, and now we get
+
+\begin{align*}
+  p(x) &= -\frac{3x^5}{8} + \frac{5x^3}{8} - \frac{15x}{8}
```
Comment (Andy-Berkshire) [inline Misc/Practice/Practice.tex:61]: replace 5x^3/8 with 10x^3/8

File: Misc/Practice/Practice.tex [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+\begin{document}
+\maketitle
+
+\section*{Algebra Problem 1}
+\subsubsection*{Problem}
+The problems can be found \href{https://www.math.hkust.edu.hk/~makyli/190_2010Sp/problemBk.pdf}{here} on page 7.
+
+\subsubsection*{Solution}
+This is a rather brute force solution, I guess better solution exists.
```
Comment (AoiYamada) [inline Misc/Practice/Practice.tex:18]: Looks like Chinese Remainder Theorem problem
here is the CRT approach:

Given
p - 1 = 0 (mod (x + 1)^3)

get
p = 1 + g(x)(x + 1)^3

Given
p + 1 = 0 (mod (x - 1)^3)
and substitute

get
1 + g(x)(x + 1)^3 + 1 = (mod (x - 1)^3)
g(x)(x + 1)^3 = -2 (mod (x - 1)^3)
g(x)(x + 1)^3 + h(x)(x - 1)^3 = -2

(x + 1)^3 & (x - 1)^3 are relatively prime

by WolframAlpha solve (use extended Euclidean algorithm, I don't know what's going on)
PolynomialExtendedGCD[ (x + 1)^3, (x-1)^3 ]

get
g(x) = -2(96x^2 - 288x + 256)/512
h(x) = -2(-96x^2 - 288x - 256)/512

by substitution
p
= 1 + g(x)(x + 1)^3
= 1 - 2(96x^2 - 288x + 256)(x + 1)^3/512
= (-3x^5 + 10x^3 - 15x)/8 

verified by symbolab
https://www.symbolab.com/solver/polynomial-long-division-calculator/long%20division%20%5Cfrac%7B%5Cfrac%7B-3x%5E%7B5%7D%2B10x%5E%7B3%7D-15x%7D%7B8%7D-1%7D%7B%5Cleft(x%2B1%5Cright)%5E%7B3%7D%7D?or=input
https://www.symbolab.com/solver/polynomial-long-division-calculator/long%20division%20%5Cfrac%7B%5Cfrac%7B-3x%5E%7B5%7D%2B10x%5E%7B3%7D-15x%7D%7B8%7D%2B1%7D%7B%5Cleft(x-1%5Cright)%5E%7B3%7D%7D?or=input

Yet, I think it is a kind of brute force too LOL

Comment (ceciliachan1979) [reply Misc/Practice/Practice.tex:18]:
```
Euclidean algorithm:

x^3 + 3x^2 + 3x + 1 = 1           (x^3 - 3x^2 + 3x - 1) + 6x^2 + 2
x^3 - 3x^2 + 3x - 1 = (x/6 - 1/2) (6x^2 + 2)            + 8x/3
6x^2 + 2            = (9x/4)      8x/3                  + 2

Extended:

From the last line, we have:
   2 = (6x^2 + 2) - (9x/4)(8x/3)

Moving up, trying to introduce (x^3 - 3x^2 + 3x - 1) and eliminate 8x/3
   2 = (6x^2 + 2) - (9x/4)((x^3 - 3x^2 + 3x - 1) - (x/6 - 1/2) (6x^2 + 2))
     = (6x^2 + 2) - (9x/4)(x^3 - 3x^2 + 3x - 1) + (9x/4)(x/6 - 1/2) (6x^2 + 2)
     = (3x^2/8 - 9x/8 + 1) (6x^2 + 2) - (9x/4)(x^3 - 3x^2 + 3x - 1) 

Moving up, trying to introduce (x^3 + 3x^2 + 3x + 1 = 1) and eliminate 6x^2 + 2
   2 = (3x^2/8 - 9x/8 + 1) ((x^3 + 3x^2 + 3x + 1) - (x^3 - 3x^2 + 3x - 1)) - (9x/4)(x^3 - 3x^2 + 3x - 1) 
     = (3x^2/8 - 9x/8 + 1) ((x^3 + 3x^2 + 3x + 1) - (x^3 - 3x^2 + 3x - 1)) - (9x/4)(x^3 - 3x^2 + 3x - 1) 
     = (3x^2/8 - 9x/8 + 1) (x^3 + 3x^2 + 3x + 1) - (3x^2/8 + 9x/8 + 1) (x^3 - 3x^2 + 3x - 1)
```

**Pull request merged**